oci/namespaces.go

```diff
@@ -4,13 +4,9 @@ import specs "github.com/opencontainers/runtime-spec/specs-go"
 
 // RemoveNamespace removes the `nsType` namespace from OCI spec `s`
 func RemoveNamespace(s *specs.Spec, nsType specs.NamespaceType) {
-	idx := -1
 	for i, n := range s.Linux.Namespaces {
 		if n.Type == nsType {
-			idx = i
+			s.Linux.Namespaces = append(s.Linux.Namespaces[:i], s.Linux.Namespaces[i+1:]...)
```

Comment:
This should work:
```Go
	for i := 0; i < len(namespaces); i++ {
		if namespaces[i] == nsType {
			namespaces = append(namespaces[:i], namespaces[i+1:]...)
		}
	}
```
It's because it's not a good idea to modify the slice you're ranging through :)